crates/outline_panel/src/outline_panel.rs

```diff
@@ -2468,7 +2550,18 @@ impl OutlinePanel {
                         return;
                     }
                     let change_focus = event.down.click_count > 1;
-                    outline_panel.toggle_expanded(&clicked_entry, window, cx);
+
+                    // Check if this entry is already selected
+                    let is_already_selected = outline_panel
+                        .selected_entry()
+                        .map(|selected| selected == &clicked_entry)
+                        .unwrap_or(false);
+
+                    // Only toggle expansion if already selected and the item is expandable
```

Comment:
Nice behavior change, let's add a small test for that?
I think we can modify one of the existing new test cases.